### DISCUSSION TITLE
Can it be used in node js or with django app?

### DISCUSSION THREAD

**Discussion by OnlinePage** [Ant-Brain/EfficientWord-Net#36] [Q&A]
Hi I want to know whether is it possible to use library with node js or with django web app?🤔 @aman-17

**Comment by TheSeriousProgrammer** [comment]
Can you explain the use case of it? Hotword detection is mainly targeted to be run on edge device rather than the server

**Comment by OnlinePage** [comment]
first of all  again @TheSeriousProgrammer your library wonderful. Now what i am trying to do is make use of same in an web app,
More specifically something like `snowboy` or `procupine` (it supports node js), therefore i asked either in django is it possible?

This library has full potenital of `procupine` already, so it would be better if we can use it in web app too, isn't it🤔?.
So what are the possible way for now as per above use case?
Also packaging an simple hotword app with pyinstaller turned out to be 1-1.5 GB in size😔

**Reply by TheSeriousProgrammer** [reply]
Thanks for the kind words, the recent model released is a onnx model, meaning it can be run inside a web browser https://onnxruntime.ai/docs/tutorials/web/

My knowledge in frontend developement is extremely limited, so pulling of this one would be a bit difficult for me.

The problem with running the model in the backend is that you will have to continuously stream your mic audio from frontend to backend, that's the most trickiest part technologically and also interms of user's privacy

Instead you can potentially run the model in the browser itself!

**Reply by OnlinePage** [reply]
`onnx` is cool. Little difficult is to re write the enitre python library from scracth in node. 
And ya running model in the backend is not good idea, even it will be no accurate as internet connection can not all time stable.
Also as u said user privacy is the major concern.😨

Any references or guide for me for node conversion of this module?🤔
